### DISCUSSION TITLE
allow_null is not fully supported in BooleanField

### DISCUSSION THREAD

**Discussion by math-a3k** [encode/django-rest-framework#8269] [Potential Issue]
Current code for BooleanField does not fully supports for null values as the default inital is set to False. This makes unsuitable for replacing NullBooleanField as it is stated in the deprecation warning.

How to reproduce:
```
from rest_framework import serializers
from django.conf import settings
from django.http import QueryDict
settings.configure()

class TestSerializer(serializers.Serializer):
    archived = serializers.BooleanField(allow_null=True)
serializer = TestSerializer(data=QueryDict(''))
assert serializer.is_valid()
assert serializer.validated_data == {'archived': None}
```
The proposed fix is here: https://github.com/math-a3k/django-rest-framework/commit/7af79a02231197cd07cadb1b4a3b66efbfe4d039
and should be ready to merge / pull request if you deem so

**Comment by auvipy** [comment]
can you please check https://github.com/encode/django-rest-framework/pull/7882 ? is this or part of it still relevant?

**Comment by jerinpetergeorge** [comment]
I think the result can be achieved by using **`NullBooleanField`** with `default=None`

<pre><code>
class TestSerializer(serializers.Serializer):
    archived = serializers.<b>NullBooleanField(default=None)</b>


serializer = TestSerializer(data=QueryDict(''))
assert serializer.is_valid()
assert serializer.validated_data == {'archived': None}
</code></pre>

**Reply by auvipy** [reply]
NullBooleanField is deprecated.

**Reply by jerinpetergeorge** [reply]
Oops, I missed that.
